nautobot/docs/development/best-practices.md

```diff
@@ -2,6 +2,11 @@
 
 While there are many different development interfaces in Nautobot that each expose unique functionality, there are a common set of a best practices that have broad applicability to users and developers alike. This includes elements of writing Jobs, Plugins, and scripts for execution through the `nbshell`.
 
+## Base Classes
+
+- All FilterSet classes should inherit from either `nautobot.utilities.filters.BaseFilterSet` (for bare-bones models) or `nautobot.extras.filters.NautobotFilterSet` (for models that support both change-logging and custom fields).
+- All Serializer classes should inherit from `ValidatedModelSerializer`.
+
```

Comment:
Can we add a note about overloading `non_filter_params` and best practices for that, seen here: https://github.com/nautobot/nautobot/pull/1794/files#diff-292c4e491fa8d2a6b999f52af45f55576703202fa9bd731438c815c0af769af5R396-R400

As well as the `get` method.